core/lib/rom/registry.rb

```diff
@@ -12,12 +12,39 @@ class Registry
     include Enumerable
     include Dry::Equalizer(:elements)
 
-    param :elements, default: -> { Hash.new }
+    param :elements
+
+    option :cache, reader: true, default: -> { Cache.new }
+
+    def self.new(*args)
```

Comment:
the problematic part is that both `elements` and `options` are hashes, so when you don't pass anything, then default elements and options become a single hash, which ends up with an empty registry with `options[:elements]`. Maybe this helps.